intro/intro_tutorial.py

```diff
@@ -0,0 +1,327 @@
+
+# %%
+from snorkel.labeling.lf import labeling_function
+
+@labeling_function()
+def lf_keyword_my(x):
+    """Many spam comments talk about 'my channel', 'my video', etc."""
+    return SPAM if "my" in x.text.lower() else ABSTAIN
+
+
+# %% [markdown]
+# Labeling functions can also use **regular expressions**:
+
+# %%
+import re
+
+@labeling_function()
+def lf_regex_check_out(x):
+    """Spam comments say 'check out my video', 'check it out', etc."""
+    return SPAM if re.search(r"check.*out", x.text, flags=re.I) else ABSTAIN
+
+
+# %% [markdown]
+# Labeling functions can of course flexibly express a range of other **heuristics**:
+
+# %%
+@labeling_function()
+def lf_short_comment(x):
+    """Ham comments are often short, such as 'cool video!'"""
+    return HAM if len(x.text.split()) < 5 else ABSTAIN
+
+
+# %% [markdown]
+# Labeling functions can also use **third-party models**:
+
+# %%
+from textblob import TextBlob
+
+@labeling_function()
+def lf_textblob_polarity(x):
+    """
+    We use a third-party sentiment classification model, TextBlob,
+    combined with the heuristic that ham comments are often positive.
+    """
+    sentiment = TextBlob(x.text).sentiment
+    return HAM if sentiment.polarity > 0.3 else ABSTAIN
+
+
+# %% [markdown]
+# For many more types of labeling functions---including over data modalities beyond text---see the other [tutorials](https://github.com/snorkel-team/snorkel-tutorials) and examples at [snorkel.org](http://snorkel.org).
+# In general the process of developing labeling functions is, like any other development process, an iterative one that takes time- but that, in [many cases](http://snorkel.org), can be orders-of-magnitude faster that hand-labeling training data.
+# For more detail on the process of developing labeling functions and other training data operators in Snorkel, see the [full version of this tutorial](https://github.com/snorkel-team/snorkel-tutorials/tree/master/spam).
+#
+# For now, we proceed by loading the above labeling functions plus some additional similar ones from a utility file:
+
+# %%
+from lfs import lfs
+
+# %% [markdown]
+# ## STEP 2: Combining & Cleaning the Labels
+#
+# Our next step is to apply the labeling functions we wrote to the unlabeled training data; we do this using the `LFApplier` corresponding to our base data class (in this case, the `PandasLFApplier`):
+
+# %%
+from snorkel.labeling.apply import PandasLFApplier
+
+applier = PandasLFApplier(lfs)
+L_train = applier.apply(df_train)
+L_dev = applier.apply(df_dev)
+L_test = applier.apply(df_test)
+
+# %% [markdown]
+# The result of applying the labeling functions (LFs) to the data is (for each split of the data) a sparse _label matrix_ with rows corresponding to data points, and columns corresponding to LFs.
+# We can take a look at some statistics of the LFs on the dev set:
+
+# %%
+from snorkel.labeling.analysis import LFAnalysis
+LFAnalysis(L_dev, lfs).lf_summary(Y=Y_dev)
```

Comment:
One of the feedback comments we got was adding kwargs to function calls. We've done that in the other longer spam tutorial. I don't know if we should also have that in the 30 second tutorial.

Reply:
Yeah I think we're now headed towards cutting down the 101 tutorial so leave it for the spam one, but agreed re: putting it there